sunpy/map/maputils.py

```diff
@@ -124,6 +124,17 @@ def sample_at_coords(smap, coordinates):
     return smap.data[smap.wcs.world_to_array_index(coordinates)]
 
 
+def _edge_coordinates(smap):
+    # Calculate all the edge pixels
+    edges = map_edges(smap)
+    edge_pixels = list(chain.from_iterable([edges[0], edges[1], edges[2], edges[3]]))
+    x = [p[0] for p in edge_pixels] * u.pix
+    y = [p[1] for p in edge_pixels] * u.pix
```

Comment:
as @ayshih pointed out on #4446 doing this comprehension twice is a waste, also this is just a `zip` (I think)

```suggestion
    x, y = list(zip(edge_pixels)) * u.pix
```